packages/ros2-test/integration/ros2__test_msgs.cpp

```diff
@@ -39,6 +35,244 @@
+    auto mock_sub = [&](const soss::Message& msg)
+    {
+      std::unique_lock<std::mutex> lock(mock_sub_mutex);
+      if(is_msg_received)
+        return;
+
+      is_msg_received = true;
+      msg_promise.set_value(msg);
+    };
+    REQUIRE(soss::mock::subscribe("transmit_arrays", mock_sub));
+
+    const std::size_t N = 3;
+    Arrays ros2_msg = generate_random_array(N);
+
+    publisher->publish(ros2_msg);
+
+    executor.spin_node_some(ros2);
+
+    // Keep spinning while we wait for the promise to be delivered. Try cycle
+    // this for no more than a few seconds. If it's not finished by that time,
+    // then something is probably broken with the test or with soss, and we
+    // should quit instead of waiting for the future and potentially hanging
+    // forever.
+    auto start_time = std::chrono::steady_clock::now();
+    while(std::chrono::steady_clock::now() - start_time < 30s)
+    {
+      executor.spin_node_some(ros2);
+      if(msg_future.wait_for(100ms) == std::future_status::ready)
+        break;
+
+      publisher->publish(ros2_msg);
+    }
+
+    REQUIRE(msg_future.wait_for(0s) == std::future_status::ready);
+    soss::Message received_msg = msg_future.get();
+
+    CHECK(received_msg.type == "test_msgs/Arrays");
+
+    std::vector<soss::Message>* data =
+        received_msg.data["basic_types_values"]
+        .cast<std::vector<soss::Message>>();
+    REQUIRE(data);
+    REQUIRE(data->size() == N);
+
+    for(std::size_t i = 0; i < N; ++i)
+    {
+      const soss::Message& soss_basic_types = (*data)[i];
+      const BasicTypes& ros2_basic_types = ros2_msg.basic_types_values[i];
+
+      #define SOSS_REQUIRE_AND_COMPARE(field_name) { \
+        const auto field_it = soss_basic_types.data.find(#field_name); \
+        REQUIRE(field_it != soss_basic_types.data.end()); \
+        BasicTypes::_ ## field_name ## _type field; \
+        soss::Convert<BasicTypes::_ ## field_name ## _type>::from_soss_field(field_it, field); \
+        CHECK(field == ros2_basic_types.field_name); }
+
+      SOSS_REQUIRE_AND_COMPARE(bool_value);
+      SOSS_REQUIRE_AND_COMPARE(byte_value);
+      SOSS_REQUIRE_AND_COMPARE(char_value);
+      SOSS_REQUIRE_AND_COMPARE(int8_value);
+      SOSS_REQUIRE_AND_COMPARE(int16_value);
+      SOSS_REQUIRE_AND_COMPARE(int32_value);
+      SOSS_REQUIRE_AND_COMPARE(int64_value);
+      SOSS_REQUIRE_AND_COMPARE(uint8_value);
+      SOSS_REQUIRE_AND_COMPARE(uint16_value);
+      SOSS_REQUIRE_AND_COMPARE(uint32_value);
+      SOSS_REQUIRE_AND_COMPARE(uint64_value);
+      SOSS_REQUIRE_AND_COMPARE(float32_value);
+      SOSS_REQUIRE_AND_COMPARE(float64_value);
+    }
+
+    const std::size_t M = ros2_msg.basic_types_values.max_size()+2;
+    data->resize(M);
+    // Add entries beyond the limits of the bounded vector so that we can make
+    // sure the message conversion is robust to overflow.
+    for(std::size_t i=N; i < M; ++i)
+      (*data)[i] = generate_random_basic_types_msg(445+i);
```

Comment:
```suggestion
      (*data)[i] = generate_random_basic_types_msg(445 + i);
```
To follow the convention.

Reply:
done